pkg/odo/cli/describe/binding.go

```diff
@@ -0,0 +1,177 @@
+package describe
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/redhat-developer/odo/pkg/api"
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/machineoutput"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+)
+
+// BindingRecommendedCommandName is the recommended binding sub-command name
+const BindingRecommendedCommandName = "binding"
+
+var describeBindingExample = ktemplates.Examples(`
+# Describe the bindings in the current devfile
+%[1]s
+
+# Describe a binding in the cluster
+%[1]s --name frontend
+`)
+
+type BindingOptions struct {
+	// nameFlag of the component to describe, optional
+	nameFlag string
+
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// working directory
+	contextDir string
+}
+
+// NewBindingOptions returns new instance of BindingOptions
+func NewBindingOptions() *BindingOptions {
+	return &BindingOptions{}
+}
+
+func (o *BindingOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *BindingOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+	if o.nameFlag == "" {
+		o.contextDir, err = os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline).NeedDevfile(""))
+		if err != nil {
+			return err
+		}
+		// this ensures that the namespace set in env.yaml is used
+		o.clientset.KubernetesClient.SetNamespace(o.GetProject())
+		return nil
+	}
+	return nil
+}
+
+func (o *BindingOptions) Validate() (err error) {
+	return nil
+}
+
+func (o *BindingOptions) Run(ctx context.Context) error {
+	if len(o.nameFlag) == 0 {
```

Comment:
Nothing wrong, but in the same file we are once checking using `o.nameFlag=='"` and next time using `len(o.nameFlag)==0`. I understand they are the same thing.